### ISSUE TITLE
use ** in auto routing

### ISSUE THREAD

**Issue by MatthiasChometon** [sveltejs/kit#9726]
### Describe the problem

I'm always frustrated when i want to do modular routing like this for example :
![image](https://user-images.githubusercontent.com/46484598/233307096-ebe76172-fe01-4b1b-b2d6-61a31073d053.png)

Because i have to add all routes in my svelte.config.js

### Describe the proposed solution

I would like svelte kit to auto generate routing with ** like this for example : src/**/routes

### Importance

would make my life easier

### Additional Information

I am currently working on svelte/kit routing to add the feature. But oh ma gad the code is not really nice and not in ts i want to cry lol. But can i do the feature and create a mr ?

Thanks in advance and thanks for your time :)

**Comment by MatthiasChometon** [comment]
i want something like this :

```js
import adapter from '@sveltejs/adapter-auto';
import { vitePreprocess } from '@sveltejs/kit/vite';

/** @type {import('@sveltejs/kit').Config} */
const config = {
	preprocess: vitePreprocess(),
	kit: {
		adapter: adapter(),
		files: {
			routes: 'src/**/routes'
		}
	},
};

export default config;
```
I want something like this :

![2023-04-20_13h03_14](https://user-images.githubusercontent.com/46484598/233347469-8de4631b-9573-4b15-8f68-15a5f0992984.png)

I want pages (routes) inside my modules. Inmy modules i can have other modules or components for example. Sorry it was disappointing to change the code, the function is so loooooong lol. I just need to finish the test and i will be done with the coe inside svelte/kit.

**Comment by dummdidumm** [comment]
I still don't fully understand. Do you mean that you want to skip `src/routes` as the root for all routes and instead place all routes inside `src` directly?

**Comment by MatthiasChometon** [comment]
Yes. And i want to have a pattern for auto generated routes in modules using routes.

**Comment by dummdidumm** [comment]
I don't understand what you mean by "pattern for auto generated routes" - what is meant by "auto generated"? Do you have a code generator somewhere that populates your folders?

**Comment by Conduitry** [comment]
It sounds like the request is to let routes be created by things matching a given glob, rather than them being created from all stuff within a given folder. If that's what the request is, it doesn't sound like a good feature to me.

**Comment by MatthiasChometon** [comment]
Yes [Conduitry](https://github.com/Conduitry). "It sounds like the request is to let routes be created by things matching a given glob, rather than them being created from all stuff within a given folder".

" If that's what the request is, it doesn't sound like a good feature to me." : Could you please develop that?

**Comment by dummdidumm** [comment]
The rules inside the route directly are: Everything with a `+` in front is a special file to SvelteKit, and folder path ~= url. We think it's important to not deviate from these rules between projects so switching between them you'll feel right at home. Since the rule is "everything with a `+` in front is a special file to SvelteKit" you can still colocate all your other stuff (graphql folder, components folder etc) in there, you just can't create a `pages` directory. What you _can_ do though is use the grouping feature and instead if `pages` you just do `(pages)` and you'll get your desired folder structure. Read more about it here: https://kit.svelte.dev/docs/advanced-routing#advanced-layouts-group

Closing since your request can be solved with existing API.

**Comment by MatthiasChometon** [comment]
> The rules inside the route directly are: Everything with a `+` in front is a special file to SvelteKit, and folder path ~= url. We think it's important to not deviate from these rules between projects so switching between them you'll feel right at home. Since the rule is "everything with a `+` in front is a special file to SvelteKit" you can still colocate all your other stuff (graphql folder, components folder etc) in there, you just can't create a `pages` directory. What you _can_ do though is use the grouping feature and instead if `pages` you just do `(pages)` and you'll get your desired folder structure. Read more about it here: https://kit.svelte.dev/docs/advanced-routing#advanced-layouts-group
> 
> Closing since your request can be solved with existing API.

You right, i should not name "pages" but what if i want to have a folder routes inside all of my modules ?
